adb_graphics/figures/skewt.py

```diff
@@ -554,111 +576,213 @@ def thermo_variables(self):
+            },
+            "pwtr": {  # Precipitable water
+                "decimals": 1,
+                "level": "sfc",
+            },
+        }
 
         for var, items in thermo.items():
-
-            varname = items.get('variable', var)
-            lev = items.get('level', 'ua')
+            varname = items.get("variable", var)
+            lev = items.get("level", "ua")
             spec = self.spec.get(varname, {}).get(lev)
 
             if not spec:
-                raise errors.NoGraphicsDefinitionForVariable(varname, lev)
+                raise errors.NoGraphicsDefinitionForVariableError(varname, lev)
 
             try:
-                tmp = self.values(level=lev, name=varname, one_lev=True)
-
-
-                transforms = spec.get('transform')
+                vals = self.values(level=lev, name=varname)
+                transforms = spec.get("transform")
                 if transforms:
-                    tmp = self.get_transform(transforms, tmp)
+                    vals = self.get_transform(transforms, vals)
 
             except errors.GribReadError:
-
-                tmp = '--'
-
-            thermo[var]['data'] = tmp
-            thermo[var]['units'] = spec.get('unit')
+                vals = DataArray([])
+            thermo[var]["data"] = vals
+            thermo[var]["units"] = spec.get("unit")
 
         return thermo
 
     def _title(self):
-
-        ''' Creates standard annotation for a skew-T. '''
+        """Creates standard annotation for a skew-T."""
 
         atime = self.date_to_str(self.anl_dt)
         vtime = self.date_to_str(self.valid_dt)
 
         # Top Left
-        plt.title(f"{self.model_name}: {atime}\nFcst Hr: {self.fhr}",
-                  fontsize=16,
-                  loc='left',
-                  position=(-4.8, 1.03),
-                  )
+        plt.title(
+            f"{self.model_name}: {atime}\nFcst Hr: {self.fhr}",
+            fontsize=16,
+            loc="left",
+            x=-4.8,
+            y=1.03,
+        )
 
         # Top Right
-        plt.title(f"Valid: {vtime}",
-                  fontsize=16,
-                  loc='right',
-                  position=(-0.20, 1.03),
-                  )
+        plt.title(
+            f"Valid: {vtime}",
+            fontsize=16,
+            loc="right",
+            x=-0.20,
+            y=1.03,
+        )
 
         # Center
         site = f"{self.site_code} {self.site_num} {self.site_name}"
         site_loc = f"{self.site_lat},  {self.site_lon}"
         site_title = f"{site} at nearest grid pt over land {site_loc}"
-        plt.title(site_title,
-                  fontsize=12,
-                  loc='center',
-                  position=(-2.5, 1.0),
-                  )
+        plt.title(
+            site_title,
+            fontsize=12,
+            loc="center",
+            x=-2.5,
+            y=1.0,
+        )
+
+
+def label_line(ax: Axes, label: str, segment: np.ndarray, **kwargs):
```

Comment:
These were previously in utils, but are only used for the SkewT, so I moved them closer to their only callers.